src/aqua/installation-spell/src/aqua/cli.aqua

```diff
@@ -13,52 +13,66 @@ func upload_deploy(config: LocalAppDeployConfig) -> DeployedAppWorkers:
     deployed <- deploy(def)
     <- deployed
 
-data LogDirect:
+data LogsDirectResult:
+    success: bool
+    error: ?string
     spell_id: string
     worker_id: string
     host_id: string
     worker_name: string
     logs: []Log
 
-func get_logs(app_workers: DeployedAppWorkers) -> *LogDirect:
-    logs: *LogDirect
+func get_logs(app_workers: DeployedAppWorkers) -> []LogsDirectResult:
+    logs: *LogsDirectResult
     for w <- app_workers.workers:
         for i <- w.installation_spells:
             on i.worker_id via i.host_id:
                 Spell i.spell_id
                 get_res <- Spell.get_logs()
-                logs <<- LogDirect(spell_id = i.spell_id, worker_id = i.worker_id, host_id = i.host_id, logs = get_res.logs, worker_name = w.name)
+                logs <<- LogsDirectResult(success = get_res.success, error = ?[get_res.error], spell_id = i.spell_id, worker_id = i.worker_id, host_id = i.host_id, logs = get_res.logs, worker_name = w.name)
 
     <- logs
 
-data LogDeal:
-    spell_id: string
-    worker_id: string
+data LogsDeal:
+    success: bool
+    error: ?string
```

Comment:
is it possible to not have separate success and error and just have optional error. If error doesn't exist - then it's a success?

Reply:
yeah, i think I can do that